pkg/awsutils/awsutils.go

```diff
@@ -248,10 +248,7 @@ func (ss *StringSet) Set(items []string) {
 }
 
 func (ss *StringSet) IsEmpty() bool {
-	if ss.data.Len() == 0 {
-		return true
-	}
-	return false
+	return ss.data != nil && ss.data.Len() == 0
 }
 
 func (ss *StringSet) Difference (other *StringSet) *StringSet {
```

Comment:
I think this method needs to hold the `other` lock too.  In this simple case/type, I think that's fine, but be aware that could lead to fun deadlocks if we get more complicated interactions between two StringSets.

Reply:
Fwiw, I think several of the uses of Difference below could instead use a regular (non-threadsafe) `sets.String`, and avoid thinking about locking.

Reply:
Make sense.  I thought about using `sets.String` but after several discussions, I went with `StringSet` approach which I thought was much cleaner.

Reply:
> I think this method needs to hold the `other` lock too. In this simple case/type, I think that's fine, but be aware that could lead to fun deadlocks if we get more complicated interactions between two StringSets.

Added `other.RLock`